static/src/js/components/SystemPermissions/SystemPermissions.jsx

```diff
@@ -55,49 +77,292 @@ const SystemPermissions = () => {
+
+                return
+              }
+
+              console.log(conceptId, '[update]', target, 'remove group from group permission', permissions)
+
+              // Remove this group from the group permission
+              mutations.push(
+                generateUpdateMutation(
+                  conceptId,
+                  groupPermissions.filter((gp) => gp.group_id !== id),
+                  target
+                )
+              )
+
+              return
+            }
+
+            // The permissions don't match CMR
+            console.log(conceptId, '[update]', target, permissions)
+
+            // Update the group permission
+            mutations.push(
+              generateUpdateMutation(
+                conceptId,
+                [
+                  ...groupPermissions.filter((gp) => gp.group_id !== id),
+                  {
+                    permissions,
+                    group_id: id
+                  }
+                ],
+                target
+              )
+            )
+
+            return
+          }
+
+          // The permissions match CMR
+          console.log(conceptId, '[no change]', target, permissions)
+
+          return
+        }
+
+        // This group does not exist in the Acl for this target in CMR
+        if (permissions.length > 0) {
+          console.log(conceptId, '[update]', target, 'add this group to the group permission', permissions)
+
+          // Add this group to the group permission
+          mutations.push(generateUpdateMutation(
+            conceptId,
+            [
+              ...groupPermissions,
+              {
+                permissions,
+                group_id: id
+              }],
+            target
+          ))
+
+          return
+        }
+      }
+
+      // This target does not have an Acl in CMR
+      if (permissions.length > 0) {
+        console.log('[create]', target, permissions)
+
+        // Create a new Acl
+        mutations.push(generateCreateMutation([{
+          permissions,
+          group_id: id
+        }], target))
+      }
+    })
+
+    const results = await Promise.allSettled(
+      mutations.map(({ mutation, variables }) => mutation({ variables }))
+    )
+
+    refetch()
+
+    const totalMutations = results.length
+    const successfulMutations = results.filter((result) => result.status === 'fulfilled')
+    const failedMutations = results.filter((result) => result.status === 'rejected')
+
+    if (totalMutations === successfulMutations.length) {
+      addNotification({
+        message: 'System permissions updated successfully.',
+        variant: 'success'
+      })
+    } else if (totalMutations === failedMutations.length) {
+      addNotification({
+        message: 'Failed to update system permissions.',
+        variant: 'danger'
+      })
+    } else {
+      addNotification({
+        message: 'Not all system permissions saved successfully, please check the form and try again.',
```

Comment:
I think we should update this to:
"Some permissions failed to update. Please check the form and try again."